evm/src/test/java/org/hyperledger/besu/evm/precompile/P256VerifyPrecompiledContractTest.java

```diff
@@ -87,4 +95,119 @@ void testP256Verify(final TestCase testCase) {
   void sanityCheck() {
     assertEquals(6_900L, contract.gasRequirement(Bytes.wrap(new byte[128])));
   }
+
+  @Test
+  void testInvalidCurvePointReturnsInvalid() {
+    // Test using a realistic attack scenario:
+    // Create r,s values that would verify against an invalid public key point
+    // The point is in the field but NOT on the secp256r1 curve
+
+    String messageHash = "44acf6b7e36c1342c2c5897204fe09504e1e2efb1a900377dbc4e7a6a133ec56";
+
+    // Point in field but NOT on curve: these coordinates are valid field elements
+    // but do not satisfy the secp256r1 curve equation y² = x³ - 3x + b
+    String invalidPubKeyX = "79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798";
+    String invalidPubKeyY = "b7c52588d95c3b9aa25b0403f1eef75702e84bb7597aabe663b82f6f04ef2777";
+
+    // Generate r,s values that could theoretically verify with this invalid point
+    // r component from a valid signature
+    String r = "c6047f9441ed7d6d3045406e95c07cd85c778e4b8cef3ca7abac09b95c709ee5";
+    // s component that would correspond to this scenario
+    String s = "30dae23890abb63e378e003d7f1d5006ab23cc7b3b65b3d0c7b45c7e1e2e08b9";
+
+    String input = messageHash + r + s + invalidPubKeyX + invalidPubKeyY;
+    Bytes inputBytes = Bytes.fromHexString(input);
+
+    // call default  implementation:
+    secp256R1.disableNative();
+    var pureJavaResult = contract.computeDefault(inputBytes);
```

Comment:
This still uses the native signature right, because the contract construction passes a native signing algorithm? Does the disableNative call one line above decide which verification pathway is chosen?

Reply:
It passes an explicit signing algorithm, which we disable native support for (disabling besu-native-ec).  

The useNativeBoringSSL config is static, so I was reluctant to modify it in tests, and instead here I just call computeDefault() which bypasses the boringSSL check and uses the configured (native disabled) signatureAlgorithm.